### PR TITLE
Update Devcontainer

### PR DESCRIPTION
**Pull request by anchpop**
Basically there are a couple changes I wanted to make

1) Use one of microsoft's images, because they're more convenient (the prompt is prettier, they come with git so you don't need to waste time adding it, etc.)

3) I added some default extensions. Microsoft's [Liveshare](https://visualstudio.microsoft.com/services/live-share/), [indent-rainbow](https://marketplace.visualstudio.com/items?itemName=oderwat.indent-rainbow), and of course vscode-unison for syntax highlighting of .u files. If you don't know what Liveshare is, I highly recommend checking it out because it's incredibly useful. I think you might find it useful when helping alpha testers debug their issues.

4) Updated the dockerfile to pull release `M2h` instead of `M2`. If there's a link that you can set up that always points to the latest release, that would be even better, because it would help ensure everyone using this devcontainer stays up-to-date.

### CODE REVIEW

**Comment by aryairani** [comment]
We should do 4). 
/cc @stew in case you have an easy way

**Comment by stew** [comment]
wow, I didn't know about this thing. Yes we have a way to always get the current release, it's always: https://github.com/unisonweb/unison/releases/tag/latest

**Comment by aryairani** [comment]
Oh wait, but latest “release”? Or latest successful build? Which one do we want here?

**Comment by anchpop** [comment]
Updated to use @stew 's link

I think using the latest release makes more sense, since this is linked in the getting-started docs which for other platforms links to releases, but I can always add an option to enable using the latest build if someone finds or sets up a link that's always up to date

**Comment by fwip** [comment]
The alias in .bashrc `ucm --codebase /path/to/codebase` seems to conflict with the `--codebase-create` flag to create a new UCM repository - I worked around by doing `$(which ucm) --codebase-create ./devCodebase`

**Comment by anchpop** [comment]
@fwip I'm not having that issue - could you be working from an old version of the PR?

**Comment by aryairani** [comment]
Okay hi all.

I still haven't been successful at running the vscode-devcontainer on aarm64 macs (docker and/or linux issues), so I've been limited in what I can test out.

I did run it on an x86 iMac just now and was able to launch `ucm` alright, but it fails the base-autodownload version parser thingie /cc @ChrisPenner, @rlmark 

```
  👋 Welcome to Unison!
  
  You are running version: 5358f82

  Get started:
  
    📖 Type help to list all commands, or help <cmd> to view help for one command
    🎨 Type ui to open the Codebase UI in your default browser
    📚 Read the official docs at https://unisonweb.org/docs
    🌎 Visit Unison Share at https://share.unison-lang.org to discover libraries
    👀 I'm watching for changes to .u files under /workspaces/vscode-devcontainer

.>
```

That said, @anchpop, apart from the "no newline at end of file" warnings, it looks good? Maybe?

File: .devcontainer/Dockerfile
Comment (aryairani) [inline .devcontainer/Dockerfile]:
```suggestion
    && rm $UCM_DIR/$UCM_ARCHIVE_NAME
```

File: .devcontainer/devcontainer.json
Comment (aryairani) [inline .devcontainer/devcontainer.json]:
```suggestion
}

```

File: .gitignore
Comment (aryairani) [inline .gitignore]:
```suggestion
*.u

```

**Review by aryairani — APPROVED** [review]

File: .devcontainer/Dockerfile
Comment (anchpop) [reply .devcontainer/Dockerfile]:
```suggestion
    && rm $UCM_DIR/$UCM_ARCHIVE_NAME

```

**Comment by anchpop** [comment]
@aryairani haven't been able to take time to figure out arm64 issues, but I don't think this PR introduces any that weren't there before haha. (I haven't tested it so I might be missing something). The newlines at EOFs should be resolved now though

**Comment by aryairani** [comment]
@anchpop Sorry for the confusion; I don't this the PR introduces any either; I just meant that I can't easily test out the devcontainer PRs anymore until the Haskell community solves arm64, so I'm counting on you! :)

**Comment by aryairani** [comment]
> Alias `ucm` to `ucm --codebase /workspace/vscode-devcontainer/devCodebase`.

@anchpop Is this actually in the PR? I couldn't spot it in the diff.

**Comment by anchpop** [comment]
@aryairani No it's not, thanks for the catch, I removed that because I thought it would be better to avoid magic

**Comment by aryairani** [comment]
> Updated to use @stew 's link
> 
> I think using the latest release makes more sense, since this is linked in the getting-started docs which for other platforms links to releases, but I can always add an option to enable using the latest build if someone finds or sets up a link that's always up to date

@stew Could you confirm, is that link the latest trunk build, or the latest named release? I think it’s the former but that we want the latter.  (Or @anchpop maybe we just hard code a link for the latter and update it manually each time?)

@anchpop In any case you are welcome to merge whenever you’re ready if you don’t want to change anything further.

**Comment by anchpop** [comment]
@aryairani I can't merge it because I don't have write access to the repo :D

**Pull request merged**
